test/ParserTests.hs

```diff
@@ -313,8 +314,18 @@ unitTests = testGroup "Unit tests"
   , testCase "test automatic open close lambda 7" $ do
       res <- runTelomareParser (parseLambda <* scn <* eof) "\\a -> (a, (\\a -> (a,0)))"
       fromRight TZero (validateVariables [] res) `compare` expr2 @?= EQ
+  , testCase "test if tictactoe.tel compiles" $ do
+      res :: Either SomeException () <- try tictactoe
+      case res of
+        Left err -> assertFailure . show $ err
+        Right _ -> pure ()
   ]
 
+tictactoe :: IO ()
+tictactoe = do
+  preludeString <- Strict.readFile "Prelude.tel"
+  Strict.readFile "tictactoe.tel" >>= runMain preludeString
```

Comment:
This will not only try to compile tictactoe, but also run it... so the test will wait for the tictactoe game to complete if it successfully compiles.

A better test would be to replace evalLoop with one iteration of main with a predefined input, and check to make sure no exception was thrown

Reply:
This works when ran in `cabal repl`, but still not when ran within a test suite:

```haskell
tictactoe :: IO String
tictactoe = do
  (Just stdin_hdl, Just stdout_hdl, m_stderr_hdl, p_hdl) <- createProcess
                                               (shell ("nix --extra-experimental-features \"nix-command flakes\" run .#telomare -- tictactoe.tel"))
                                               { std_in = CreatePipe
                                               , std_out = CreatePipe
                                               }
  hPutStrLn stdin_hdl "1"
  hPutStrLn stdin_hdl "4"
  hPutStrLn stdin_hdl "2"
  hPutStrLn stdin_hdl "5"
  hPutStrLn stdin_hdl "3"

  hClose stdin_hdl
  res <- hGetContents stdout_hdl
  putStrLn "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
  putStrLn res
  let aux = lines res
      !winner = aux !! (length aux - 2)
  cleanupProcess (Just stdin_hdl, Just stdout_hdl, m_stderr_hdl, p_hdl)
  pure winner
  -- pure "hola"
```

The output (when not in a test-suite) is:

```
*Main
λ> tictactoe 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
warning: Git tree '/home/hhefesto/src/telomare' is dirty
1|2|3
-+-+-
4|5|6
-+-+-
7|8|9
Player 2's turn
please input number of square: 
O|2|3
-+-+-
4|5|6
-+-+-
7|8|9
Player 1's turn
please input number of square: 
O|2|3
-+-+-
X|5|6
-+-+-
7|8|9
Player 2's turn
please input number of square: 
O|O|3
-+-+-
X|5|6
-+-+-
7|8|9
Player 1's turn
please input number of square: 
O|O|3
-+-+-
X|X|6
-+-+-
7|8|9
Player 2's turn
please input number of square: 
Player 2wins!
done

"Player 2wins!"
*Main
λ> 
```

When added to the test suite it complains with:

```
warning: you don't have Internet access; disabling some network-dependent features
error: creating directory '/nix/var': Permission denied
```

working on getting this resolved